src/Modules/Reports/Commands.Reports/NewPowerBIReport.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+using Microsoft.PowerBI.Common.Client;
+using Microsoft.PowerBI.Common.Api.Reports;
+using Microsoft.PowerBI.Common.Api.Workspaces;
+using Microsoft.PowerBI.Common.Abstractions;
+
+namespace Microsoft.PowerBI.Commands.Reports
+{
+    [Cmdlet(CmdletVerb, CmdletName)]
+    [OutputType(typeof(Report))]
+    public class NewPowerBIReport : PowerBIClientCmdlet
+    {
+        public const string CmdletVerb = VerbsCommon.New;
+        public const string CmdletName = "PowerBIReport";
+
+        #region Parameters
+
+        [Parameter(Position = 0, Mandatory = true)]
+        public string Path { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public string Name { get; set; }
+
+        [Alias("GroupId")]
+        [Parameter(Mandatory = false)]
+        public Guid WorkspaceId { get; set; }
+
+        [Alias("Group")]
+        [Parameter(Mandatory = false)]
+        public Workspace Workspace { get; set; }
+        #endregion
+
+        #region Constructors
+        public NewPowerBIReport() : base() { }
+
+        public NewPowerBIReport(IPowerBIClientCmdletInitFactory init) : base(init) { }
+        #endregion
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.Workspace != null)
+            {
+                this.WorkspaceId = this.Workspace.Id;
```

Comment:
I think you should enforce separate parameter sets so that a user can't accidentally overwrite a provided workspace ID with one from a workspace.